src/Data/Rational/Properties.agda

```diff
@@ -614,6 +614,15 @@ toℚᵘ-isOrderMonomorphism-< = record
 <-asym : Asymmetric _<_
 <-asym (*<* p<q) (*<* q<p) = ℤ.<-asym p<q q<p
 
+<-dense : Dense _<_
+<-dense {p} {q} p<q with m , p<ᵘm , m<ᵘq ← ℚᵘ.<-dense (toℚᵘ-mono-< p<q)
```

Comment:
Should this not just be a let statement? Do we really need `with`?

Reply:
You're absolutely correct (and I've been a demon for insisting on it elsewhere, ... :facepalm:)

Reply:
Blast! No, actually: If I (try to) use `let`, the the let-bound names don't scope correctly over the subsequent `where` clauses... which is very annoying. Or am I missing something?

Ideally, I would make a simultaneous definition in the `where` block` and destruct it in place, as for example
```agda
<-dense {p} {q} p<q = fromℚᵘ m , p<m , m<q
  where
  d : ∃[ m ] p < m × m < q
  d@(m , p<ᵘm , m<ᵘq) = ℚᵘ.<-dense (toℚᵘ-mono-< p<q)
  
  m≃m : m ≃ᵘ toℚᵘ (fromℚᵘ m)
  m≃m = ℚᵘ.≃-sym (toℚᵘ-fromℚᵘ m)
  
  p<m = toℚᵘ-cancel-< (ℚᵘ.<-respʳ-≃ m≃m p<ᵘm)
  
  m<q = toℚᵘ-cancel-< (ℚᵘ.<-respˡ-≃ m≃m m<ᵘq)
```
but that doesn't seem to be legal, either. Using/Having to use projections would be an abomination.

Reply:
If you know a better way, I'd love to see it!

Reply:
Hmm weird. You're right I can't see any good way of working around that. I agree that what you've written with the `where` block above would be the "correct" way of doing it in my opinion but I can't get that to work either. So I guess we just fall back on good old `let`:
```
<-dense : Dense _<_
<-dense {p} {q} p<q =
  let
    m , p<ᵘm , m<ᵘq = ℚᵘ.<-dense (toℚᵘ-mono-< p<q)

    m≃m : m ≃ᵘ toℚᵘ (fromℚᵘ m)
    m≃m = ℚᵘ.≃-sym (toℚᵘ-fromℚᵘ m)

    p<m : p < fromℚᵘ m
    p<m = toℚᵘ-cancel-< (ℚᵘ.<-respʳ-≃ m≃m p<ᵘm)

    m<q : fromℚᵘ m < q
    m<q = toℚᵘ-cancel-< (ℚᵘ.<-respˡ-≃ m≃m m<ᵘq)
  in fromℚᵘ m , p<m , m<q
```

Reply:
I think this is a little clearer, so I've pushed the change.